src/lightning/fabric/fabric.py

```diff
@@ -404,22 +402,16 @@ def backward(self, tensor: Tensor, *args: Any, model: Optional[_FabricModule] =
             model as argument here.
 
         """
-        module = model._forward_module if model is not None else model
-        if isinstance(self._strategy, DeepSpeedStrategy):
-            if model is None:
-                if self._models_setup == 0:
-                    raise RuntimeError("No models were set up for backward. Did you forget to call `fabric.setup()`?")
-                if self._models_setup > 1:
-                    raise ValueError(
-                        "When using multiple models + deepspeed, please provide the model used to perform"
-                        " the optimization: `self.backward(loss, model=model)`"
-                    )
-                module = self._strategy.model
-            else:
-                # requires to attach the current `DeepSpeedEngine` for the `_FabricOptimizer.step` call.
-                self._strategy._deepspeed_engine = module
-
-        self._strategy.backward(tensor, module, *args, **kwargs)
+        # rank_zero_deprecation(
+        #    "`fabric.backward(loss)` is no longer necessary, you can simply do `loss.backward()` as with pure PyTorch",
+        #    stacklevel=6,
+        # )
```

Comment:
We could deprecate `fabric.backward`. `_BackwardTensor` is used silently anyways. If decided, it can be done in a follow-up too.